src/Compilers/Core/Portable/Emit/CommonPEModuleBuilder.cs

```diff
@@ -456,16 +450,14 @@ private SynthesizedDefinitions GetCacheOfSynthesizedDefinitions(TNamedTypeSymbol
             {
                 return _synthesizedDefs.GetOrAdd(container, _ => new SynthesizedDefinitions());
             }
-            else
-            {
-                SynthesizedDefinitions defs;
-                if (!_synthesizedDefs.TryGetValue(container, out defs))
-                {
-                    defs = null;
-                }
 
-                return defs;
+            SynthesizedDefinitions defs;
+            if (!_synthesizedDefs.TryGetValue(container, out defs))
+            {
+                defs = null;
```

Comment:
It's reasonable to remove the assignment and if. There are many examples of that in the code.